src/random/secure.cr

```diff
@@ -1,15 +1,29 @@
 require "crystal/system/random"
 
-# Generates random numbers from a secure source provided by the system.
+# `Random::Secure` generates random numbers from a secure source provided by the system.
 #
-# For example `arc4random` is used on OpenBSD, whereas on Linux it uses
-# `getrandom` (if the kernel supports it) and fallbacks on reading from
-# `/dev/urandom` on UNIX systems.
+# It uses a [cryptographically secure pseudorandom number generator (CSRPNG)](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator)
+# for cryptography and secure usages such as generating secret keys, or seeding
+# a pseudorandom number generator (PRNG).
 #
 # ```
 # Random::Secure.rand(6)            # => 4
 # [1, 5, 6].shuffle(Random::Secure) # => [6, 1, 5]
 # ```
+#
+# On OpenBSD, it uses [`arc4random`](https://man.openbsd.org/arc4random),
+# on Linux [`getrandom`](http://man7.org/linux/man-pages/man2/getrandom.2.html) (if the kernel supports it),
+# on Windows [`RtlGenRandom`](https://docs.microsoft.com/en-us/windows/win32/api/ntsecapi/nf-ntsecapi-rtlgenrandom),
+# and falls back to reading from `/dev/urandom` on UNIX systems.
+#
+# For generating *high quality* random numbers, a pseudorandom number generator
```

Comment:
I'd rather remove this whole paragraph. "high quality" is pretty confusing and undefined.